src/workspaceUpdater.ts

```diff
@@ -0,0 +1,45 @@
+import * as vscode from "vscode";
+import { BINARY_UPDATE_WORKSPACE_INTERVAL } from "./globals/consts";
+import sendUpdateWorkspaceRequest from "./binary/requests/workspace";
+import { Logger } from "./utils/logger";
+
+const INITIAL_DELAY = 3000;
+
+let updateInterval: NodeJS.Timeout | null = null;
+
+export default function startWorkspaceUpdater() {
+  setTimeout(() => {
```

Comment:
you need to start the loop after the binary is ready, see https://github.com/codota/tabnine-vscode/blob/77451b9438459ed8db2e56e13a40e3d9dde432e9/src/authentication/TabnineAuthenticationProvider.ts#L89 for exmple